regulations/urls.py

```diff
@@ -27,156 +27,153 @@
+
     # Redirect to a diff based on GET params
     # Example http://.../diff_redirect/201-3/old_version?new_version=new
-    url(r'^diff_redirect/%s/%s$' % (section_pattern, version_pattern),
+    url(r'^diff_redirect/(?P<label_id>[\d]+[-][\w]+)/(?P<version>[-\d\w_]+)$',
         diff_redirect, name='diff_redirect'),
+
     # A section by section paragraph with chrome
     # Example: http://.../sxs/201-2-g/2011-1738
-    url(r'^sxs/%s/%s$' % (paragraph_pattern, notice_pattern),
-        lt_cache(ChromeSXSView.as_view()),
-        name='chrome_sxs_view'),
+    url(r'^sxs/(?P<label_id>[-\w]+)/(?P<notice_id>[-\d\w_]+)$',
+        lt_cache(ChromeSXSView.as_view()), name='chrome_sxs_view'),
+
     # Search results for non-JS viewers
     # Example: http://.../search?q=term&version=2011-1738
-    url(r'^search(?:/cfr)?/%s$' % reg_pattern,
-        ChromeSearchView.as_view(), name='chrome_search',
+    url(r'^search(?:/cfr)?/(?P<label_id>[\d]+)$',
+        ChromeSearchView.as_view(),
+        name='chrome_search',
         kwargs={'doc_type': 'cfr'}),
-    url(r'^search/preamble/%s$' % preamble_pattern,
-        ChromePreambleSearchView.as_view(), name='chrome_search_preamble',
+    url(r'^search/preamble/(?P<label_id>[\w]+)$',
+        ChromePreambleSearchView.as_view(),
+        name='chrome_search_preamble',
         kwargs={'doc_type': 'preamble'}),
+
     # Diff view of a section for non-JS viewers (or book markers)
     # Example: http://.../diff/201-4/2011-1738/2013-10704
-    url(r'^diff/%s/%s/%s$' %
-        (section_pattern, version_pattern, newer_version_pattern),
-        lt_cache(ChromeSectionDiffView.as_view()),
-        name='chrome_section_diff_view'),
-
+    url(r'^diff/(?P<label_id>[\d]+[-][\w]+)/(?P<version>[-\d\w_]+)/(?P<newer_version>[-\d\w_]+)$',
+        lt_cache(ChromeSectionDiffView.as_view()), name='chrome_section_diff_view'),
     url(r'^preamble/(?P<doc_number>[\w-]+)/cfr_changes/(?P<section>[\w-]+)$',
         daily_cache(CFRChangesView.as_view()), name='cfr_changes'),
     url(r'^preamble/(?P<paragraphs>[-\w]+(/[-\w]+)*)$',
         daily_cache(PreambleView.as_view()), name='chrome_preamble'),
 
     # Redirect to version by date
     # Example: http://.../201-3-v/1999/11/8
-    url(r'^%s/(?P<year>\d{4})/(?P<month>\d{2})/(?P<day>\d{2})$'
-        % paragraph_pattern, redirect_by_date, name='redirect_by_date'),
+    url(r'^(?P<label_id>[-\w]+)/(?P<year>\d{4})/(?P<month>\d{2})/(?P<day>\d{2})$',
+        redirect_by_date, name='redirect_by_date'),
+
     # Redirect to version by current date
     # Example: http://.../201-3-v/CURRENT
-    url(r'^%s/CURRENT$' % paragraph_pattern,
+    url(r'^(?P<label_id>[-\w]+)/CURRENT$',
         redirect_by_current_date, name='redirect_by_current_date'),
+
     # A regulation section with chrome
     # Example: http://.../201-4/2013-10704
-    url(r'^%s/%s$' % (section_pattern, version_pattern),
+    url(r'^(?P<label_id>[\d]+[-][\w]+)/(?P<version>[-\d\w_]+)$',
```

Comment:
I find this one particularly hard to differentiate from a few of it's peers, like the one from line 105
```
url(r'^(?P<label_id>[\d]+)/(?P<version>[-\d\w_]+)$',
```
I'm not sure what the options are, but in this example the difference between matching `123` vs `123-something` is what would need to stand out.

Reply:
one option might be a middle-ground between this and the original pattern.. Something like
```
url(r'^<label_id>%s/<version>%s$' % (match_section, match_version),
```